enterprise/web.jsf/src/org/netbeans/modules/web/jsf/wizards/JSFClientGenerator.java

```diff
@@ -1016,45 +1014,43 @@ public void run(CompilationController compilationController) throws IOException
-                    "int p = 0;\n" +
-                    "int grabStart = 0;\n" +
-                    "String delim = \"#\";\n" +
-                    "String escape = \"~\";\n" +
-                    "Pattern pattern = Pattern.compile(escape + \"*\" + delim);\n" +
-                    "Matcher matcher = pattern.matcher(string);\n" +
-                    "while (matcher.find()) {\n" +
-                    "String found = matcher.group();\n" +
-                    "if (found.length() % 2 == 1) {\n" +
-                    "params[p] = string.substring(grabStart, matcher.start());\n" +
-                    "p++;\n" +
-                    "grabStart = matcher.end();\n" +
-                    "}\n" +
-                    "}\n" +
-                    "if (p != params.length - 1) {\n" +
-                    "throw new IllegalArgumentException(\"string \" + string + \" is not in expected format. expected " + params + " ids delimited by \" + delim);\n" +
-                    "}\n" +
-                    "params[p] = string.substring(grabStart);\n" +
-                    "for (int i = 0; i < params.length; i++) {\n" +
-                    "params[i] = params[i].replace(escape + delim, delim);\n" +
-                    "params[i] = params[i].replace(escape + escape, escape);\n" +
-                    "}\n\n"
-                    );
+            getIdBody.append("String params[] = new String[").append(params).append("];\n")
+                    .append("int p = 0;\n")
+                    .append("int grabStart = 0;\n")
+                    .append("String delim = \"#\";\n")
+                    .append("String escape = \"~\";\n")
+                    .append("Pattern pattern = Pattern.compile(escape + \"*\" + delim);\n")
+                    .append("Matcher matcher = pattern.matcher(string);\n")
+                    .append("while (matcher.find()) {\n")
+                    .append("String found = matcher.group();\n")
+                    .append("if (found.length() % 2 == 1) {\n")
+                    .append("params[p] = string.substring(grabStart, matcher.start());\n")
+                    .append("p++;\n")
+                    .append("grabStart = matcher.end();\n")
+                    .append("}\n")
+                    .append("}\n")
+                    .append("if (p != params.length - 1) {\n")
+                    .append("throw new IllegalArgumentException(\"string \" + string + \" is not in expected format. expected ").append(params).append(" ids delimited by \" + delim);\n")
+                    .append("}\n")
+                    .append("params[p] = string.substring(grabStart);\n")
+                    .append("for (int i = 0; i < params.length; i++) {\n")
+                    .append("params[i] = params[i].replace(escape + delim, delim);\n")
+                    .append("params[i] = params[i].replace(escape + escape, escape);\n")
+                    .append("}\n\n");
```

Comment:
i am not a big fan of those kind of changes. Readability wise, both are similar, but performance wise the original version is going to win. Simple concatenations with constants "foo"+"bar" becomes "foobar" in bytecode anyway, rest is done via invoke dynamic and is likely to be faster than append.

Looking forward, it is also easier to convert concatenations to text blocks in future, which would have a readability gain without performance penalty. Or even later to string templates. append chains are a step back in that regard.

sb.append() is fine for loops or situations where the String structure is not known.

Reply:
True.

My initial concern was for the readabiility of the automatic cleanup. Future features for the moment are that, future features. Until then my preference is that I'd rather read a multiline version of code/pseudo code like this instead of a single line string hundreds of characters long.

Had we had textblocks available I'd have preferred that even over the current state of the change, but we don't.

Reply:
so then why change it? Is it a cleanup when there is no improvement in readability while having a performance cost (even though performance doesn't matter at all in this particular case, its not in a inner loop or anything like that).

Reply:
lets agree at least to not run the append refactoring for future cleanups - its a legacy code inspection from java 1.5 times. 

But it would be good if the big text block candidates could be reverted back to the original version, so that they can be easier converted in future (NB has a refactoring for that but not for append chains).